src/algebra/lie/nilpotent.lean

```diff
@@ -67,6 +78,32 @@ class is_nilpotent : Prop :=
 instance trivial_is_nilpotent [is_trivial L M] : is_nilpotent R L M :=
 ⟨by { use 1, change ⁅⊤, ⊤⁆ = ⊥, simp, }⟩
 
+lemma nilpotent_endo_of_nilpotent_module [hM : is_nilpotent R L M] :
+  ∃ (k : ℕ), ∀ (x : L), (to_endomorphism R L M x)^k = 0 :=
+begin
+  unfreezingI { obtain ⟨k, hM⟩ := hM, },
+  use k,
+  intros x, ext m,
+  rw [linear_map.pow_apply, linear_map.zero_apply, ← @lie_submodule.mem_bot R L M, ← hM],
+  exact iterated_action_mem_lower_central_series R L M x m k,
+end
+
+/-- For a nilpotent Lie module, the weight space of the 0 weight is the whole module.
+
+This result will be used downstream to show that weight spaces are Lie submodules, at which time
+it will be possible to state it in the language of weight spaces. -/
+lemma zero_weight_space_eq_top_of_nilpotent [is_nilpotent R L M] :
```

Comment:
Should the name mention `max_gen_eigenspace` instead of `weight_space`, given that you intend to restate/reformulate it later.

(Aside: I think `maximal_generalized_eigenspace` should be renamed to something shorter, like `max_gen_eigenspace`.)

Reply:
Agreed. I have renamed to `infi_max_gen_zero_eigenspace_eq_top_of_nilpotent`.

I actually have plans to do some light refactoring of the eigenspace stuff so I will rename `maximal_generalized_eigenspace` to `max_gen_eigenspace` as part of that.